test/hotspot/jtreg/serviceability/monitoring/ThreadInfo/getLockOwnerName/getLockOwnerName.java

```diff
@@ -0,0 +1,415 @@
+
+            if (name == null) {
+                out.println("Failure at " + count + " loops.");
+                throw new RuntimeException("ThreadInfo.getLockOwnerName() "
+                                           + "returned null name for "
+                                           + "contender.");
+            } else if (!name.equals("blocker")) {
+                out.println("Failure at " + count + " loops.");
+                throw new RuntimeException("name='" + name + "': name "
+                                           + "should be blocker.");
+            } else {
+                logDebug("ThreadInfo.getLockOwnerName() returned blocker.");
+            }
+
+            synchronized (barrierReleaser) {
+                // tell releaser thread to release the blocker thread
+                testState = TS_READY_TO_RELEASE;
+                barrierReleaser.notify();
+            }
+
+            while (true) {
+                // maxDepth == 0 requires no safepoint so alternate.
+                int maxDepth = ((count % 1) == 1) ? Integer.MAX_VALUE : 0;
+                info = mbean.getThreadInfo(id, maxDepth);
+                if (info == null) {
+                    // the contender has exited
+                    break;
+                }
+                name = info.getLockOwnerName();
+                // We can't verify that name == null here because contender
+                // might be slow leaving the threadLock monitor.
+                lateCount++;
+            }
+            logDebug("made " + lateCount + " late calls to getThreadInfo() " +
+                     "and info.getLockOwnerName().");
+
+            try {
+                releaser.join(JOIN_MAX * 1000);
+                if (releaser.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("releaser thread is stuck");
+                }
+                blocker.join(JOIN_MAX * 1000);
+                if (blocker.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("blocker thread is stuck");
+                }
+                contender.join(JOIN_MAX * 1000);
+                if (contender.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("contender thread is stuck");
+                }
+            } catch (InterruptedException ex) {
+            }
+
+            checkTestState(TS_CONTENDER_DONE);
+        }
+
+        System.out.println("Executed " + count + " loops in " + timeMax +
+                           " seconds.");
+
+        return 0;
+    }
+}
+
+class getLockOwnerNameWorker extends Thread {
+    public getLockOwnerNameWorker(String name) {
+        super(name);
+    }
+
+    public void run() {
+        getLockOwnerName.logDebug("thread running");
+
+        //
+        // Launch the blocker thread:
```

Comment:
The comment says "Launch the blocker thread", but this thread is already launched. Maybe drop "Launch" in favor of "Run", or just say "The block thread". Same comment for the other two threads.